src/components/VulnerabilityEdgePrompt.js

```diff
@@ -1,39 +1,174 @@
 import React from 'react';
-import {
-  Button,
-  EmptyState,
-  EmptyStateBody,
-  EmptyStateIcon,
-  Title,
-} from '@patternfly/react-core';
+import { useContext, useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { MonitoringIcon } from '@patternfly/react-icons';
-import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import RegistryContext from '../store/registeryContext';
+import { Alert, Button } from '@patternfly/react-core';
+import AsyncComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+import { getDevice, getImageSet } from '../api/edge/imagesInfo';
+import PropTypes from 'prop-types';
 
+import { InProgressIcon } from '@patternfly/react-icons';
 const VulnerabilityEdgePrompt = () => {
-  const chrome = useChrome();
+  const { getRegistry } = useContext(RegistryContext);
   const { inventoryId } = useParams();
+  const [CVEs, setCVEs] = useState(null);
+  const [imageId, setImageID] = useState(null);
+  const [imageSetId, setImageSetID] = useState(null);
+  const [deviceData, setDeviceData] = useState(null);
+  const [newImageStatus, setNewImageStatus] = useState(null);
+  useEffect(() => {
+    if (CVEs?.data?.length > 0) {
+      const data = getActiveAlert(CVEs, deviceData, newImageStatus, imageId);
+      setActiveAlert(data);
+    }
+  });
+  useEffect(() => {
+    (async () => {
+      const a = await getDevice(inventoryId);
+      setDeviceData(a);
+      setImageID(a?.ImageInfo?.Image?.ID);
+      setImageSetID(a?.ImageInfo?.Image?.ImageSetID);
+      const params = {
+        id: 'view',
+        q: { id: imageSetId },
+      };
+      const resp = await getImageSet(params);
+      setNewImageStatus(resp?.data?.[0]?.Status);
+    })();
+  }, []);
+  const [activeAlert, setActiveAlert] = useState('noAlert');
+
+  const getActiveAlert = (CVEs, deviceData, newImageStatus, imageId) => {
+    if (!CVEs?.data?.length > 0 || !imageId) {
+      return 'noAlert';
+    }
+    if (
+      deviceData?.UpdateTransactions[0]?.Status === 'BUILDING' ||
+      deviceData?.UpdateTransactions[0]?.Status === 'CREATED'
+    ) {
+      return 'systemUpdating';
+    }
+    if (deviceData?.Device?.UpdateAvailable) {
+      return 'updateDevice';
+    }
+    if (newImageStatus === 'BUILDING' || newImageStatus === 'CREATED') {
+      return 'imageBuilding';
+    }
+    return 'updateImage';
+  };
+
+  // useEffect(() => {
+  //   !CVEs?.isLoading &&
+  //     !CVEs?.meta?.filter &&
+  //     setUpdateCveModal((prevState) => ({
+  //       ...prevState,
+  //       cveCount: CVEs?.data?.length,
+  //     }));
+  // }, [CVEs, deviceData, newImageStatus, imageId]);
+
+  const alerts = {
```

Comment:
These could be in their own component files instead of defining them within another component.